msgs/src/main/scala/com/thenewmotion/ocpi/msgs/v2_1/Commands.scala

```diff
@@ -1,48 +1,136 @@
-package com.thenewmotion.ocpi.msgs.v2_1
+package com.thenewmotion.ocpi.msgs
+package v2_1
 
 import java.time.ZonedDateTime
+import java.util.UUID
 
-import com.thenewmotion.ocpi.msgs.Url
-import com.thenewmotion.ocpi.msgs.v2_1.Locations.{ConnectorId, EvseUid, LocationId}
-import com.thenewmotion.ocpi.msgs.v2_1.Sessions.SessionId
-import com.thenewmotion.ocpi.msgs.v2_1.Tokens.Token
+import Locations.{ConnectorId, EvseUid, LocationId}
+import Sessions.SessionId
+import Tokens.Token
 import com.thenewmotion.ocpi.{Enumerable, Nameable}
 
 object Commands {
-  sealed trait CommandType extends Nameable
-  object CommandType extends Enumerable[CommandType] {
-    case object ReserveNow extends CommandType {val name = "RESERVE_NOW"}
-    case object StartSession extends CommandType {val name = "START_SESSION"}
-    case object StopSession extends CommandType {val name = "STOP_SESSION"}
-    case object UnlockConnector extends CommandType {val name = "UNLOCK_CONNECTOR"}
-    val values = Seq(ReserveNow, StartSession, StopSession, UnlockConnector)
+
+  sealed trait CommandName extends Nameable
+  object CommandName extends Enumerable[CommandName] {
+    case object ReserveNow extends CommandName {val name = "RESERVE_NOW"}
+    case object StartSession extends CommandName {val name = "START_SESSION"}
+    case object StopSession extends CommandName {val name = "STOP_SESSION"}
+    case object UnlockConnector extends CommandName {val name = "UNLOCK_CONNECTOR"}
+    val values = Iterable(ReserveNow, StartSession, StopSession, UnlockConnector)
+  }
+
+  abstract class Command(val name: CommandName) {
+    def responseUrl: Url
+  }
+
+  private[ocpi] def callbackUrl(
+    baseUrl: Url,
+    cmdName: CommandName,
```

Comment:
Why do you need a command name here? Uuid should be enough.

Reply:
True, the examples given in the OCPI docs had both the command name and an identifier so I went with that....

Reply:
That's it then, other that this.